Client/game_sa/CAutomobileSA.cpp

```diff
@@ -11,9 +11,79 @@
 
 #include "StdInc.h"
 #include "CAutomobileSA.h"
+#include "CGameSA.h"
+
+extern CGameSA* pGame;
 
 CAutomobileSA::CAutomobileSA(CAutomobileSAInterface* pInterface)
 {
     SetInterface(pInterface);
     Init();
 }
+
+void CAutomobileSAInterface::SetPanelDamage(std::uint8_t panelId, bool breakGlass, bool spawnFlyingComponent)
+{
+    int nodeId = CDamageManagerSA::GetCarNodeIndexFromPanel(panelId);
+    eCarNodes node = static_cast<eCarNodes>(nodeId);
+
+    RwFrame* frame = m_aCarNodes[nodeId];
+    if (!frame)
+        return;
+
+    CVehicleModelInfoSAInterface* vehicleInfo = nullptr;
+    if (auto* mi = pGame->GetModelInfo(m_nModelIndex))
+        vehicleInfo = static_cast<CVehicleModelInfoSAInterface*>(mi->GetInterface());
+
+    if (!vehicleInfo || !vehicleInfo->IsComponentDamageable(nodeId))
+        return;
+
+    switch (m_damageManager.GetPanelStatus(panelId))
+    {
+        case DT_PANEL_DAMAGED:
+        {
+            if ((pHandlingData->uiModelFlags & 0x10000000) != 0) // check bouncePanels flag
+                return;
+
+            if (node != eCarNodes::WINDSCREEN && node != eCarNodes::WING_LF && node != eCarNodes::WING_RF)
+            {
+                // Get free bouncing panel
+                CBouncingPanelSAInterface* panelInterface = nullptr;
+                for (auto& panel : m_panels)
+                {
+                    if (panel.m_nFrameId == static_cast<std::uint16_t>(-1))
+                    {
+                        panelInterface = &panel;
+                        break;
+                    }
+                }
+
+                panelInterface->SetPanel(nodeId, 1, GetRandomNumberInRange(-0.2f, -0.5f));
+            }
+
+            SetComponentVisibility(frame, 2); // ATOMIC_IS_DAM_STATE
+            break;
+        }
+        case DT_PANEL_OPENED:
+        {
+            if (panelId == WINDSCREEN_PANEL)
+                m_VehicleAudioEntity.AddAudioEvent(91, 0.0f);
+
+            SetComponentVisibility(frame, 2); // ATOMIC_IS_DAM_STATE
+            break;
+        }
+        case DT_PANEL_OPENED_DAMAGED:
+        {
+            if (panelId == WINDSCREEN_PANEL)
+            {
+                if (breakGlass)
+                    ((void(__cdecl*)(CAutomobileSAInterface*))0x71C2B0)(this); // Call CGlass::CarWindscreenShatters
```

Comment:
Doesn't the function require two arguments?
![image](https://github.com/user-attachments/assets/e8bf3987-1cfe-413e-890d-7fcc044cc40b)

Reply:
In my IDB and on gta-reversed, there is only one argument.

Reply:
Even in the Android release there is a bool argument, and the assembly code in the picture is pretty clear.

Reply:
This function is only called in ``CAutomobile::SetPanelDamage``, and no additional argument is passed there either. The signature is probably incorrect, or the argument itself is unused, though the implementation on gta-reversed also doesn't include this argument. I think we can safely ignore it.

https://github.com/gta-reversed/gta-reversed/blob/dbe5054cb3b9e59566d23f5d768d84c8478b9b19/source/game_sa/Glass.cpp#L81

Reply:
.